impeller/geometry/gradient.cc

```diff
@@ -16,14 +16,21 @@ static void AppendColor(const Color& color, std::vector<uint8_t>* colors) {
   colors->push_back(converted[3]);
 }
 
-std::vector<uint8_t> CreateGradientBuffer(const std::vector<Color>& colors,
-                                          const std::vector<Scalar>& stops,
-                                          uint32_t* out_texture_size) {
+GradientData CreateGradientBuffer(const std::vector<Color>& colors,
+                                  const std::vector<Scalar>& stops) {
+  // check for gradients that are invalid due to either not enough colors or
+  // mismatched colors and stops
+  if (colors.size() < 2 ||
+      (stops.size() != 0 && stops.size() != colors.size())) {
+    return GradientData{
+        .color_bytes = {},
+        .texture_size = 0,
+    };
```

Comment:
For the colors/stops check -- is this something that we normalize in the display list/shouldn't be possible through user error? If so, should a debug assert be added here?

Reply:
Ahh yeah, this shouldn't be possible. Made it a DCHECK